python/basix/finite_element.py

```diff
@@ -657,26 +668,56 @@ def create_custom_element(
         embedded_superdegree: Degree of a polynomial in this
             element's polyset.
         poly_type: Type of polyset to use for this element.
+        dtype: Element scalar type.
 
     Returns:
         A custom finite element.
     """
-    return FiniteElement(
-        _create_custom_element(
-            cell_type.value,
-            value_shape,
-            wcoeffs,
-            x,
-            M,
-            interpolation_nderivs,
-            map_type.value,
-            sobolev_space.value,
-            discontinuous,
-            embedded_subdegree,
-            embedded_superdegree,
-            poly_type.value,
+    if dtype is None:
+        from basix import default_scalar_type
+
+        dtype = default_scalar_type
+
+    if wcoeffs.dtype != dtype:
+        wcoeffs = np.dtype(dtype).type(wcoeffs)  # type: ignore
+        x = [[np.dtype(dtype).type(j) for j in i] for i in x]  # type: ignore
+        M = [[np.dtype(dtype).type(j) for j in i] for i in M]  # type: ignore
+    if np.issubdtype(dtype, np.float32):
+        return FiniteElement(
+            _create_custom_element_float32(
```

Comment:
I wonder if using `std::variant` in the nanobind layer could avoid the duplication with  `_create_custom_element_float32` and  `_create_custom_element_float64`.